### PR TITLE
Refactor: submit bag from body instead of headers

### PR DESCRIPTION
**Pull request by Johnny850807**
resolve: #349 

Refactor: submit bag from the body instead of headers

### CODE REVIEW

File: Spring-Boot/Spring-Boot-Submission/src/test/java/tw/waterball/judgegirl/springboot/submission/controllers/AbstractSubmissionControllerTest.java [hunk 1]
```diff
@@ -344,16 +349,22 @@ protected ResultActions submitCode(int studentId, Token token, MockMultipartFile
...
-        for (var entry : submissionBag.entrySet()) {
-            addingHeaders = call.header(HEADER_BAG_KEY_PREFIX + entry.getKey(), entry.getValue());
-        }
-        return addingHeaders;
+
+        var bagPart = new MockPart(SUBMISSION_BAG_MULTIPART_KEY_NAME, SUBMISSION_BAG_MULTIPART_KEY_NAME,
+                toJson(submissionBag).getBytes(UTF_8));
+        bagPart.getHeaders().setContentType(MediaType.APPLICATION_JSON);
+        parts.add(bagPart);
+        call.part(parts.toArray(new Part[0]));
```
Comment (Wally5077) [inline Spring-Boot/Spring-Boot-Submission/src/test/java/tw/waterball/judgegirl/springboot/submission/controllers/AbstractSubmissionControllerTest.java:365]: delete List<Part>, and change to user code below
```Java
call.part( new Part[]{bagPart});
```

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

File: Spring-Boot/Spring-Boot-Submission/src/test/java/tw/waterball/judgegirl/springboot/submission/controllers/AbstractSubmissionControllerTest.java (hunk 1 above)
Comment (Johnny850807) [reply Spring-Boot/Spring-Boot-Submission/src/test/java/tw/waterball/judgegirl/springboot/submission/controllers/AbstractSubmissionControllerTest.java:365]: `call.part(bagPart);`

**Review by Wally5077 — APPROVED** [review]
LGTM

**Pull request merged**